modular_zubbers/code/modules/mob/living/carbon/human/human.dm

```diff
@@ -1,2 +1,7 @@
 /mob/living/carbon/human/species/shadekin
 	race = /datum/species/shadekin
+
+
+// This is expected to be called or used in situations where you already know the mob is dead
+/mob/living/carbon/human/proc/get_dnr()
+	return ((HAS_TRAIT(src, TRAIT_DNR) || !((src.mind?.get_ghost(FALSE, TRUE)) ? 1 : 0)))
```

Comment:
We don't use the DNR trait really as a check, it's been super flaky, so the most accurate method to get a DNR is to determine if theres either a ckey or a ghost that can be grabbed. If neither, the person is DNR. It's more reliable as it catches respawns and ghost role taking, and other instances of permanent no client

I've edited the DNR trait to ghost people out of their mob immediately, so that if the ghost to body link is severed, it's the most reliable tell